SMT-LIB/parse_all.py

```diff
@@ -1,44 +1,45 @@
 import os
-import time
-import sys
+from timeit import timeit
 import random
 import multiprocessing
 import argparse
 
-from itertools import islice
+from itertools import islice, repeat
 
 from pysmt.shortcuts import reset_env, read_smtlib
 
 
 def get_all_smt_files(target_dir=None):
-    if target_dir == None:
+    if target_dir is None:
         target_dir = "./"
 
     assert os.path.exists(target_dir)
     for root, _, files in os.walk(target_dir):
         for f in files:
-            if f.endswith(".smt2"):
+            if f.endswith(".smt2") or f.endswith(".smt2.bz2"):
                 yield os.path.join(root, f)
 
 
-def execute_script_fname(smtfile):
+def execute_script_fname(smtfile, reps):
     """Read and call a Solver to solve the instance"""
     print(smtfile)
     reset_env()
     assert os.path.exists(smtfile)
-    start = time.clock()
-    read_smtlib(smtfile)
-    end = time.clock()
-    return ( (end - start), smtfile)
+    assert reps > 0
+    try:
+        exec_time = timeit(lambda: read_smtlib(smtfile, False), number=reps)
+    except:
+        exec_time = -1
+    return (exec_time, smtfile)
 
 
 def dump_stats(timings, fname):
     if fname is None:
         fname = "stats.out"
-    with open(fname, "w") as f:
+    with open(fname, mode='w', encoding="utf-8") as f:
         f.write('filename, time\n')
-        for k in timings:
-            f.write('%f, "%s"\n' % k)
+        for val, bench_name in timings:
+            f.write(f'"{bench_name}", {val}\n')
```

Comment:
FYI, fstrings are python >=3.6. If we decide to ignore older versions of python (and considering 3.5 is EoL I am generally positive) we should probably document this more clearly.

I think no change is required here as this is an util and not part of library itself.